Metalama.Framework/Advising/IAdviceFactory.cs

```diff
@@ -42,7 +42,14 @@ public interface IAdviceFactory
         /// <seealso href="@overriding-methods"/>
         IOverrideAdviceResult<IMethod> Override( IMethod targetMethod, in MethodTemplateSelector template, object? args = null, object? tags = null );
 
-        IIntroductionAdviceResult<INamedType> IntroduceType( INamespaceOrNamedType targetNamespaceOrType, string typeName, TypeKind typeKind, Action<INamedTypeBuilder>? buildType = null );
+        ITypeIntroductionAdviceResult IntroduceType( INamespaceOrNamedType targetNamespaceOrType, string name, TypeKind typeKind, Action<INamedTypeBuilder>? buildType = null );
+
+        ITypeIntroductionAdviceResult IntroduceType( string targetNamespace, string name, TypeKind typeKind, Action<INamedTypeBuilder>? buildType = null );
+
+        ITypeIntroductionAdviceResult IntroduceClass(
```

Comment:
Should there also be an overload for `IntroduceClass(string, ...)`?

Reply:
Yeah, I created task to finalize the API. I'm not sure I like to have overloads for string namespace, I would rather have something like ICompilation.GetNamespace that would get you INamespace/INamespaceBuilder.